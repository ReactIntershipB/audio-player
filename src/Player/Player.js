import React, { Component } from 'react';
import { Avatar, Button, Col, Row, Slider } from 'antd';

class Player extends Component {

  render() {

    let style = {
      width: '100%',
      height: '120px',
      paddingTop: '10px',
      position: 'fixed',
      bottom: '0rem',
      border: 'solid 1px lightgrey'
    };

    return (

      <div style={style}>
        
        <Row  type='flex'
              justify='center'
              align='middle'>

          <Col span={2}>

            <Avatar shape='square'
                    size={80}
                    icon='star' />

            <br />

            <span>Track 1</span>

          </Col>
          
          <Col span={20}>

            <Row  type='flex'
                  justify='center'
                  align='middle'>
              
              <Col span={2}>
              
                <Button href='#'>shuffle</Button>
              
              </Col>

              <Col span={2}>

                <Button shape='circle'
                        size={'large'}
                        icon='backward'/>
              
              </Col>

              <Col span={2}>
              
                <Button shape='circle'
                        size={'large'}
                        icon='caret-right'/>
              
              </Col>

              <Col span={2}>
              
                <Button shape='circle'
                        size={'large'}
                        icon='forward'/>
              
              </Col>

               <Col span={2}>
              
                <Button href='#'>repeat</Button>
              
              </Col>

            </Row>

            <Row  type='flex'
                  justify='center'
                  align='middle'>

              <Col span={22}>
              
                <Slider defaultValue={25} disabled={false} />
              
              </Col>

              <Col span={2}>
              
                <span>2:05</span>
              
              </Col>

            </Row>

          </Col>

        </Row>

      </div>

    );

  }

}

export default Player;
