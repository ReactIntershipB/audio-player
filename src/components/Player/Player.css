.player {
    width: 100%;
    height: 150%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.player-title {
    color: white;
    z-index: 10000;
    text-align: center;
    font-family: 'Lora', serif;
    font-size: 2em;
    text-shadow: 1px 1px 1px black;
}

.buttons-container {
    display: flex;
    align-items: center;
    justify-content: space-evenly;
    width: 60%;
    margin: 0 auto;
}

.buttons-container button {
    opacity: 0.7;
}

.slider-container {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    width: 75%;
    margin: 0 auto;
    align-items: flex-end;
}

.slider-container p {
    color: white;
    z-index: 100000;
    text-align: center;
    margin: 0;
    height: 21px;
}

.slider-duration-container {
    display: flex;
    width: 100%;
    justify-content: space-between;
}

.ant-slider.ant-slider {
    width: 100%;
    margin: 0;
}

.ant-slider-handle.ant-slider-handle,
.ant-slider-rail.ant-slider-rail,
.ant-slider-track.ant-slider-track,
.ant-slider-step.ant-slider-step {
    background-color: #D81159;
    border: solid 2px #D81159;
}

.slider-container .ant-slider:hover .ant-slider-handle:not(.ant-tooltip-open),
.slider-container .ant-slider-handle.ant-tooltip-open {
    border-color: #8F2D56;
}

.buttons-container .ant-btn:active,
.buttons-container .ant-btn.active {
    color: #8F2D56;
    border-color: #8F2D56;
}

.buttons-container .ant-btn:hover,
.buttons-container .ant-btn:focus {
    color: #D81159;
    border-color: #D81159;
}

.slider-container .ant-slider-handle:focus {
    border-color: #8F2D56;
    -webkit-box-shadow: 0 0 0 5px #8F2D56;
    box-shadow: 0 0 0 3px #8F2D56;
}
