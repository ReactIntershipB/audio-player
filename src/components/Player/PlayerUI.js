import { observable, action } from 'mobx';

export class PlayerUI {
    @observable timer = 0;
    @observable songIsPlaying = false;

    setTimer (songDuration) {
        this.songDuration = songDuration;
        this.initTimer();
    }

    @action playSong() {
        this.songIsPlaying = true;
    }

    @action pauseSong() {
        this.songIsPlaying = false;
    }

    @action reset () {
        clearInterval(this.timerInterval);
        this.timer = 0;
        this.songIsPlaying = false;
    }

    @action timerStep () {
        this.timer = this.timer + 0.1;
    }

    initTimer() {
        this.timerInterval = setInterval(() => {
            if (this.songIsPlaying) {
                if (this.timer < this.songDuration) {
                    this.timerStep();
                } else {
                    this.reset();
                }
            }
        }, 100);
    }

    @action
    updateTimer (value) {
        this.timer = value;
    }
}
