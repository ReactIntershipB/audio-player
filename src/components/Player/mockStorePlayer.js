import { observable } from 'mobx';

export const mockData = {
    data: {
        id: 3135553,
        readable: true,
        title: 'One More Time',
        title_short: 'One More Time',
        title_version: '',
        link: 'https://www.deezer.com/track/3135553',
        duration: 320,
        rank: 817122,
        explicit_lyrics: false,
        preview: 'https://cdns-preview-e.dzcdn.net/stream/c-e77d23e0c8ed7567a507a6d1b6a9ca1b-7.mp3',
        artist: {
            id: 27,
            name: 'Daft Punk',
            tracklist: 'https://api.deezer.com/artist/27/top?limit=50',
            type: 'artist'
        }
    }
};

class MockModel {
    @observable data = [];
    @observable loading = false;

    find = jest.fn(() => {
        this.getData();
    });

    getData = () => {
        this.data = mockData;
    }
}

export class SongModel extends MockModel {
    @observable currentSongId = 0;

    get songLink() {
        return this.data.preview;
    }

    get songTitle() {
        return this.data.title;
    }

    get songLength() {
        return 30;
    }

    get songLoaded() {
        return !!this.data.preview;
    }

    setCurrentSongId(id) {
        this.currentSongId = id;
    }
}

export class AppUI {
    @observable isPlaying = false;
    @observable isButtonDisabled = true;

    togglePlaying() {
        this.isPlaying = !this.isPlaying;
    }

    updatePlayingStatus(status) {
      this.isPlaying = status;
    }

    changeButtonStatus(term) {
      this.isButtonDisabled = !term;
    }
}
