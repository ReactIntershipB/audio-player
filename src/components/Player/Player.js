import React from 'react';
import PropTypes from 'prop-types';
import { Avatar, Button, Col, Row, Slider } from 'antd';
import { observer, inject } from 'mobx-react';
import './Player.css';
import { PlayIcon } from './../common/PlayIcon';
import { autorun } from 'mobx';

@inject('appUI', 'songModel')
@observer
class Player extends React.Component {
  constructor(props) {
    super(props);
    this.props.songModel.init();

    autorun(() => {
      if (this.songLink) {
        this.props.appUI.togglePause();
      }
    });
  }

  handleNextSongClick = () => {
    ;
  }

  handlePreviousSongClick = () => {
    ;
  }

  sliderChange = (value) => {
    // TO DO
  }

  secondsToStringTime = (time) => {
    return `${parseInt(time / 60)}:${time % 60}`;
  }

  get trackLength () {
    // return this.props.songModel.currentSong.duration ? this.props.songModel.currentSong.duration : 0;
    return 30;
  }

  get trackTitle () {
    return this.props.songModel.track ? this.props.songModel.track.title : '';
  }

  get songLink () {
    return this.props.songModel.currentSong.preview;
  }

  render() {
    return (
      <div className='player'>

        <audio autoPlay src={this.songLink}></audio>

        <Row type='flex'
          justify='center'
          align='middle'>

          <Col span={2}>
            <Avatar shape='square'
              size={80}
              icon='star' />

            <br />

            <span>Title</span>

          </Col>

          <Col span={20}>

            <Row
              type='flex'
              justify='center'
              align='middle'>
              <span className='title'>{this.trackTitle}</span>
            </Row>

            <Row
              type='flex'
              justify='center'
              align='middle'>

              <Col
                span={2}
                className='btns'>

                <Button>
                  <i className="fas fa-random"></i>
                </Button>
              </Col>

              <Col
                span={2}
                className='btns'>

                <Button shape='circle'
                  size={'large'}
                  icon='backward'
                  onClick={this.handlePreviousSongClick}
                />
              </Col>

              <Col
                span={2}
                className='btns'>

                <PlayIcon disabled={!this.songLink}/>

              </Col>

              <Col
                span={2}
                className='btns'>

                <Button shape='circle'
                  size={'large'}
                  icon='forward'
                  onClick={this.handleNextSongClick}
                />
              </Col>

              <Col span={2} className='btns'>
                <Button>
                  <i className="fas fa-redo-alt"></i>
                </Button>
              </Col>

            </Row>

            <Row
              type='flex'
              justify='center'
              align='middle'>

              <Col span={22}>
                <Slider min={0}
                  max={this.props.songModel.track ? this.props.songModel.track.time : 0}
                  value={this.props.appUI.timer}
                  disabled={false}
                  onChange={this.sliderChange} />
              </Col>

              <Col span={2}>
                <span>{this.trackTimeStatus}</span>
              </Col>

            </Row>

          </Col>

        </Row>
      </div>

    );
  }
}

Player.propTypes = {
  songModel: PropTypes.object,
  appUI: PropTypes.object
};

export default Player;
