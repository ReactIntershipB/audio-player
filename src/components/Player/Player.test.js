import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';

import Player from './Player';
import * as mockStore from './mockStorePlayer';

describe('Player', () => {
    it('should match snapshot', () => {
        const props = {
            albumModel: {},
            songModel: new mockStore.SongModel(),
            playerModel: {},
            appUI: {}
        };

        const player = renderer
            .create(
                <Provider
                    albumModel={props.albumModel}
                    songModel={props.songModel}
                    playerModel={props.playerModel}
                    appUI={props.appUI}
                >
                    <Player />
                </Provider>
            ).toJSON();

        expect(player).toMatchSnapshot();
    });
});
