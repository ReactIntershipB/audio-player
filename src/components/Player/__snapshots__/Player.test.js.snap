// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match snapshot 1`] = `
<div
  className="player"
  style={
    Object {
      "backgroundImage": "url(null)",
    }
  }
>
  <audio
    autoPlay={true}
    id="audioPlayer"
    src={undefined}
  />
  <h4
    className="player-general player-author"
  >
    Deezer Player
  </h4>
  <p
    className="player-general player-title"
  >
    
  </p>
  <div
    className="slider-container"
  >
    <div
      className="ant-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      style={undefined}
    >
      <div
        className="ant-slider-rail"
        style={Object {}}
      />
      <div
        className="ant-slider-track"
        style={
          Object {
            "left": "0%",
            "width": "0%",
          }
        }
      />
      <div
        className="ant-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={30}
        aria-valuemin={0}
        aria-valuenow={0}
        className="ant-slider-handle"
        onBlur={[Function]}
        onClick={undefined}
        onContextMenu={undefined}
        onFocus={undefined}
        onKeyDown={[Function]}
        onMouseDown={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={undefined}
        role="slider"
        style={
          Object {
            "left": "0%",
          }
        }
        tabIndex={0}
      />
      <div
        className="ant-slider-mark"
      />
    </div>
    <div
      className="slider-duration-container"
    >
      <p>
         
      </p>
      <p>
         
      </p>
    </div>
  </div>
  <div
    className="buttons-container"
  >
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-random"
        style={
          Object {
            "color": "grey",
          }
        }
      />
    </button>
    <button
      className="ant-btn ant-btn-circle ant-btn-lg ant-btn-icon-only"
      onClick={[Function]}
      type="button"
    >
      <i
        className="anticon anticon-backward"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="backward"
          fill="currentColor"
          height="1em"
          onClick={undefined}
          style={undefined}
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M448.98 106.1L12.68 481.6c-16.9 15.7-16.9 43.3 0 59l436.2 377.3c24.4 22.8 63.1 4.7 63.1-29.5V135.6c.1-34.1-38.6-52.2-63-29.5zM960.98 106.1l-436.3 375.5c-8.4 7.9-12.6 18.7-12.6 29.5s4.2 21.6 12.6 29.5l436.2 377.3c24.4 22.8 63.1 4.7 63.1-29.5V135.6c.1-34.1-38.6-52.2-63-29.5z"
          />
        </svg>
      </i>
    </button>
    <button
      className="ant-btn play-button ant-btn-circle ant-btn-lg ant-btn-icon-only"
      data-testid="play-button"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      <i
        className="anticon anticon-caret-right"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="caret-right"
          fill="currentColor"
          height="1em"
          onClick={undefined}
          style={undefined}
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M392.7 76.3l489.1 403.1a42.53 42.53 0 0 1 0 63.4L392.7 947.7c-27.4 24.4-70.8 5-70.8-31.7V107.9c.1-36.6 43.4-56.1 70.8-31.6z"
          />
        </svg>
      </i>
    </button>
    <button
      className="ant-btn ant-btn-circle ant-btn-lg ant-btn-icon-only"
      onClick={[Function]}
      type="button"
    >
      <i
        className="anticon anticon-forward"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="forward"
          fill="currentColor"
          height="1em"
          onClick={undefined}
          style={undefined}
          viewBox="0 0 1024 1024"
          width="1em"
        >
          <path
            d="M575.1 106.1l436.2 375.4c16.9 15.7 16.9 43.3 0 59L575.1 917.9c-24.4 22.8-63.1 4.7-63.1-29.5V135.6c0-34.1 38.7-52.2 63.1-29.5zM63.1 106.1l436.2 375.4c8.4 7.9 12.6 18.7 12.6 29.5s-4.2 21.6-12.6 29.5L63.1 917.9C38.7 940.6 0 922.5 0 888.4V135.7c0-34.2 38.7-52.3 63.1-29.6z"
          />
        </svg>
      </i>
    </button>
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <i
        className="fas fa-redo-alt"
        style={
          Object {
            "color": "grey",
          }
        }
      />
    </button>
  </div>
</div>
`;
